test/broadcast.jl

```diff
@@ -337,3 +337,15 @@ end
 @test broadcast(+, 1.0, (0, -2.0)) == (1.0,-1.0)
 @test broadcast(+, 1.0, (0, -2.0), [1]) == [2.0, 0.0]
 @test broadcast(*, ["Hello"], ", ", ["World"], "!") == ["Hello, World!"]
+
+# Ensure that even strange constructors that break `T(x)::T` work with broadcast
```

Comment:
Do we actually care about such constructors? Is is supported to return an object not of type `T`? (I've been wondering about this for one of my packages.